python/util/templates.py

```diff
@@ -110,13 +111,26 @@ def generate_react_components(self):
        newDomain.update(_success, error, baseDomainEmbed);
     }
 
-    function setJob(newJob, success, error) {
-       function _success () {
-         window.job = newJob;
-         redrawAll();
-         success();
-       }
-       newJob.update(_success, error, baseJobEmbed);
+    function updateJob(options) {
+        var jobCopy = new MERCHI.Job();
+        var embed = options.embed ? options.embed : baseJobEmbed;
+        function _success (jobData) {
+            window.job = jobData
+            redrawAll();
+            options.success();
+        }
+        jobCopy.id(window.job.id());
+        job.get(_success, options.error, embed);
+    }
+
+    function setJob(options) {
+        var embed = options.embed ? options.embed : baseJobEmbed;
+        function _success (jobData) {
+            window.job = options.responseUpdate ? jobData : options.job;
```

Comment:
no, you cannot let the one component unilaterally set the job. it must **only** come from the server, with the appropriate things also embedded, otherwise other components are interfered with.